.github/workflows/update-node-dist.yml

```diff
@@ -0,0 +1,51 @@
+name: Update Node dist
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      # implemented since 28, once branched off, add you stable branch here
+      - main
+      - ci/node-dist-workflow
+      # - stable28
+
+concurrency:
+  group: update-node-dist-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  update-node-dist:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+
+      - name: Read package.json node and npm engines version
+        uses: skjnldsv/read-package-engines-version-actions@8205673bab74a63eb9b8093402fd9e0e018663a1 # v2.1
+        id: versions
+        with:
+          fallbackNode: '^20'
+          fallbackNpm: '^9'
+
+      - name: Set up node ${{ steps.versions.outputs.nodeVersion }}
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3
+        with:
+          node-version: ${{ steps.versions.outputs.nodeVersion }}
+
+      - name: Set up npm ${{ steps.versions.outputs.npmVersion }}
+        run: npm i -g npm@"${{ steps.versions.outputs.npmVersion }}"
+
+      - name: Setup git
+        run: |
+          git config --local user.email "nextcloud-command@users.noreply.github.com"
+          git config --local user.name "nextcloud-command"
+      - name: Install dependencies & build
+        run: |
+          npm ci
+          npm run build --if-present
+      - name: Add and commit
+        run: |
+          git add --force js/
```

Comment:
Currently this will fail the CI if there is nothing to commit. See first run on this branch.
Would be nice to handle that more gracefully. Either with an `if:` condition or by making the whole job depend on changes to `src`, `package.json` and so on.

Reply:
I pushed a commit to reuse the build changes check in here for that.